packages/experimental/Badge/src/Badge.tsx

```diff
@@ -19,7 +19,8 @@ export const badgeLookup = (layer: string, userProps: BadgeProps): boolean => {
     layer === userProps['shape'] ||
     (!userProps['shape'] && layer === 'rounded') ||
     layer === userProps['color'] ||
-    (!userProps['color'] && layer === 'brand')
+    (!userProps['color'] && layer === 'brand') ||
+    (userProps.icon && layer === 'textPadding')
```

Comment:
Now that textPadding is not a state, this doesn't need to be part of the lookup.